plugins/org.obeonetwork.m2doc.sirius/src/org/obeonetwork/m2doc/sirius/SiriusDiagramByTitleProvider.java

```diff
@@ -0,0 +1,186 @@
+import org.eclipse.sirius.diagram.DSemanticDiagram;
+import org.eclipse.sirius.diagram.business.api.query.DDiagramQuery;
+import org.eclipse.sirius.diagram.ui.tools.api.part.DiagramEditPartService;
+import org.eclipse.sirius.viewpoint.DRepresentation;
+import org.eclipse.sirius.viewpoint.DView;
+import org.eclipse.sirius.viewpoint.description.AnnotationEntry;
+import org.eclipse.sirius.viewpoint.description.Viewpoint;
+import org.obeonetwork.m2doc.provider.DiagramProvider;
+import org.obeonetwork.m2doc.provider.IProvider;
+import org.obeonetwork.m2doc.provider.OptionType;
+import org.obeonetwork.m2doc.provider.ProviderConstants;
+import org.obeonetwork.m2doc.provider.ProviderException;
+
+/**
+ * {@link SiriusDiagramByTitleProvider} are used to get sirius diagrams from an
+ * eObject and a name.
+ * 
+ * @author Romain Guider
+ */
+public class SiriusDiagramByTitleProvider extends DiagramProvider {
+	/**
+	 * The key used in the map passed to {@link IProvider} to define the Sirius
+	 * representation title from which image will be computed.
+	 */
+	String KEY_REPRESENTATION_TITLE = "title";
+
+	/**
+	 * Creates a new {@link SiriusDiagramByTitleProvider} given an EObject which
+	 * is the root of the seek diagram and the name of the diagram.
+	 * 
+	 * @param eObject
+	 *            the root of the diagram
+	 * @param title
+	 *            the name of the diagram
+	 * @param projectRootPath
+	 *            the project root path
+	 */
+	public SiriusDiagramByTitleProvider() {
+
+	}
+
+	/**
+	 * Replace forbidden characters with "_" in a filename
+	 * 
+	 * @param name
+	 * @return sanitized string
+	 */
+	public static String sanitizeFilename(String name) {
+		return name.replaceAll("[:\\\\/*?|<>]", "_");
+	}
+
+	/**
+	 * Get the full path (starting from the root folder) of the generated image
+	 * for a diagram
+	 * 
+	 * @param diagram
+	 * @param rootPath the project root path from which document generation has been launched.
+	 * @return Full path starting from root folder
+	 */
+	private String getDiagramImageFilename(DSemanticDiagram diagram, String rootPath) {
+		return rootPath + "/generated/images/representations/diagram_"
+				+ sanitizeFilename(diagram.eResource().getURIFragment(diagram)) + "." + ImageFileFormat.JPEG.getName();
+	}
+
+	/**
+	 * Retrieve all representations whose target is the specified EObject
+	 * 
+	 * @param session the Sirius session from which we want to find the representation with the given name.
+	 * @return List of representations
+	 */
+	private DRepresentation getAssociatedRepresentationByName(String representationName, Session session) {
+		if (representationName == null) {
+			return null;
+		}
+		Collection<DRepresentation> representations = DialectManager.INSTANCE.getAllRepresentations(session);
+
+		// Filter representations to keep only those in a selected viewpoint
+		Collection<Viewpoint> selectedViewpoints = session.getSelectedViewpoints(false);
```

Comment:
The session might be null if the configuration isn't in a Sirius session. It's worth checking null here or in the caller method.